shaderc-sys/build/build.rs

```diff
@@ -0,0 +1,194 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+mod cmd_finder;
+
+use std::env;
+use std::env::consts;
+use std::path::{Path, PathBuf};
+
+static COMBINED_LIB: &str = "shaderc_combined";
+static COMBINED_LIB_FILE: &str = "libshaderc_combined.a";
+static SPIRV_LIB: &str = "libSPIRV.a";
+
+fn build_shaderc(shaderc_dir: &PathBuf, use_ninja: bool) -> PathBuf {
+    let mut config = cmake::Config::new(shaderc_dir);
+    config
+        .profile("Release")
+        .define("CMAKE_POSITION_INDEPENDENT_CODE", "ON")
+        .define("SPIRV_SKIP_EXECUTABLES", "ON")
+        .define("SPIRV_WERROR", "OFF")
+        .define("SHADERC_SKIP_TESTS", "ON")
+        .define("CMAKE_INSTALL_LIBDIR", "lib");
+    if use_ninja {
+        config.generator("Ninja");
+    }
+    config.build()
+}
+
+fn build_shaderc_msvc(shaderc_dir: &PathBuf) -> PathBuf {
+    let mut config = cmake::Config::new(shaderc_dir);
+    config
+        .profile("Release")
+        .define("CMAKE_POSITION_INDEPENDENT_CODE", "ON")
+        .define("SPIRV_SKIP_EXECUTABLES", "ON")
+        .define("SPIRV_WERROR", "OFF")
+        .define("SHADERC_SKIP_TESTS", "ON")
+        // cmake-rs tries to be clever on Windows by injecting several
+        // C/C++ flags, which causes problems. So I have to manually
+        // define CMAKE_*_FLAGS_* here to suppress that.
+        .define("CMAKE_C_FLAGS", " /nologo /EHsc")
+        .define("CMAKE_CXX_FLAGS", " /nologo /EHsc")
+        .define("CMAKE_C_FLAGS_RELEASE", " /nologo /EHsc")
+        .define("CMAKE_CXX_FLAGS_RELEASE", " /nologo /EHsc")
+        .define("CMAKE_INSTALL_LIBDIR", "lib")
+        .generator("Ninja");
+    config.build()
+}
+
+fn main() {
+    // Deprecated --no-defaults path inhereted from before shaderc-rs & shaderc-sys split
+    if env::var("CARGO_FEATURE_CHECK_PASSTHROUGH").is_ok()
+        && env::var("CARGO_FEATURE_DONT_USE_DEPRECATED").is_err()
+    {
+        let out_dir = env::var("OUT_DIR").unwrap();
+        println!("cargo:warning=USE OF --no-defaults IS DEPRECATED BEHAVIOR.");
```

Comment:
(could be old question) Let's back up.  The way I preserved the old behavior is so that in `shaderc-rs` you can still say:
```$ cargo test --no-default-features``` 
And it just works like it does in master.

This is done by adding the two goofy features.  I don't anticipate that anyone will explicitly do something like:
```toml
[dependencies]
shaderc-sys = {version = "0.5.0", features = ["enable-inverted-no-defaults"]
```

Basically it's definitely a shaderc-rs choice, part-and-parcel with using shaderc-rs, but has to be implemented in shaderc-sys do to the way `--no-default-features` behaves